checkbox-provider-ce-oem/tests/test_check_vpu_device.py

```diff
@@ -0,0 +1,277 @@
+
+        self.assertEqual(
+            "INFO:root:SoC family is {}".format(expected_soc_family),
+            lc.output[-1])
+        self.assertEqual(soc_family, expected_soc_family)
+
+    @mock.patch("pathlib.Path.is_file")
+    def test_soc_family_not_available(self, mock_is_file):
+        """
+        Checking SoC family file is not available
+        """
+        mock_is_file.return_value = False
+        with self.assertRaisesRegex(
+            SystemExit,
+            "{}/family file is not available".format(
+                check_vpu_device.SOC_ROOT
+            )
+        ):
+            check_vpu_device.get_soc_family()
+
+    @mock.patch("pathlib.Path.read_text")
+    @mock.patch("pathlib.Path.is_file")
+    def test_soc_id_available(self, mock_is_file, mock_read_text):
+        """
+        Checking SoC ID file is available
+        """
+        expected_soc_id = "i.MX8MQ"
+        mock_is_file.return_value = True
+        mock_read_text.return_value = expected_soc_id
+        with self.assertLogs() as lc:
+            soc_id = check_vpu_device.get_soc_id()
+
+        self.assertEqual(
+            "INFO:root:SoC ID is {}".format(expected_soc_id),
+            lc.output[-1])
+        self.assertEqual(soc_id, expected_soc_id)
+
+    @mock.patch("pathlib.Path.is_file")
+    def test_soc_id_not_available(self, mock_is_file):
+        """
+        Checking SoC ID file is not available
+        """
+        mock_is_file.return_value = False
+        with self.assertRaisesRegex(
+            SystemExit,
+            "{}/soc_id file is not available".format(
+                check_vpu_device.SOC_ROOT
+            )
+        ):
+            check_vpu_device.get_soc_id()
+
+    @mock.patch("pathlib.Path.read_text")
+    @mock.patch("pathlib.Path.is_file")
+    def test_kernel_version_available(self, mock_is_file, mock_read_text):
+        """
+        Checking kernel version file is available
+        """
+        kernel_raw_data = (
+            "Linux version 5.15.0-89-generic (buildd@lcy02-amd64-004) "
+            "(gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0, "
+            "GNU ld (GNU Binutils for Ubuntu) 2.34) "
+            "#99~20.04.1-Ubuntu SMP Thu Nov 2 15:16:47 UTC 2023"
+        )
+        expected_ver = "5.15"
+        mock_is_file.return_value = True
+        mock_read_text.return_value = kernel_raw_data
+        kernel_ver = check_vpu_device.get_kernel_version()
+
+        self.assertEqual(kernel_ver, expected_ver)
+
+    @mock.patch("pathlib.Path.read_text")
+    @mock.patch("pathlib.Path.is_file")
+    def test_kernel_version_mismatch(self, mock_is_file, mock_read_text):
+        """
+        Checking kernel version format mismatch
+        """
+        kernel_raw_data = "Linux version 5.1-89-generic (amd64-004)"
+        mock_is_file.return_value = True
+        mock_read_text.return_value = kernel_raw_data
+        with self.assertRaisesRegex(
+            SystemExit,
+            "Failed to identify kernel version"
```

Comment:
I am wondering do we really need to check the error message? Because this might change if we modify the code like adding more error handling.
I suggest that we only check if it raise `SystemExit` when there's error.

Reply:
fixed